packages/now-cli/src/util/dev/router.ts

```diff
@@ -150,7 +150,9 @@ export async function devRouter(
               if (routeConfig.status && phase === 'miss') {
                 status = routeConfig.status;
               }
-              reqPathname = destPath;
+              const destParsed = url.parse(destPath, true);
+              query = Object.assign(destParsed.query, query);
+              reqPathname = destParsed.pathname || '/';
```

Comment:
@TooTallNate @ijjk @Timer I'm not sure if we should be hitting this code at all considering `@vercel/next` has its own dev server. What do you think?

Reply:
from my observations, in this configuration `serveProjectAsNowV2` goes through the `devRouter` multiple times with varying arguments. For each phase it can go through: 
- [`initial route resolution`](https://github.com/vercel/vercel/blob/3d4261300c3311b7e8faa7bd01ba0f72bccb1fb6/packages/now-cli/src/util/dev/server.ts#L1373) (which is pre-match and thus doesn't know the url will be handled by next)
- [`findBuildMatch`](https://github.com/vercel/vercel/blob/3d4261300c3311b7e8faa7bd01ba0f72bccb1fb6/packages/now-cli/src/util/dev/server.ts#L2108) which decides if it [`shouldServe`](https://github.com/vercel/vercel/blob/3d4261300c3311b7e8faa7bd01ba0f72bccb1fb6/packages/now-cli/src/util/dev/server.ts#L2146) 
- resolve effective proxied url 

since the `pathName` is corrupted in the initial route resolution `shouldServe` never gets the right information...and ends up resolving an incorrect path, it seems to understand it must proxy it to `next` but in the process what should be a query string ends up encoded as a escaped path segment

Reply:
It seems like this case is already handled on line 181 below:

```js
const destParsed = url.parse(destPath, true);
Object.assign(destParsed.query, query);
```

Why is it needed here too? Is line 181 never executed?